macOS/unwindmacos.cpp

```diff
@@ -43,38 +43,15 @@ void UnwindDataMacOS::LookupEncoding(size_t original_address) {
   }
 }
 
-void UnwindDataMacOS::LookupLSDA(size_t original_address) {
-  if (last_lookup.IsLsdaMiss(original_address)) {
-    if (!lsda_map.empty() && (last_lookup.encoding & UNWIND_HAS_LSDA)) {
-      auto it = lsda_map.upper_bound(original_address);
-      if (it == lsda_map.begin()) {
-        last_lookup.SetLsda(-1, 0, it->first - 1);
-      } else if (it == lsda_map.end()) { // Sentinel entry
-        last_lookup.SetLsda(-1, prev(it)->first, -1);
-      } else {
-        last_lookup.SetLsda(prev(it)->second, prev(it)->first, it->first - 1);
-      }
-    } else {
-      last_lookup.SetLsda(0, last_lookup.encoding_min_address, last_lookup.encoding_max_address);
-    }
-  }
-}
-
 void UnwindDataMacOS::AddMetadata(size_t original_address, size_t translated_address) {
   LookupEncoding(original_address);
   if (!last_lookup.IsEncodingValid()) {
     return;
   }
 
-  LookupLSDA(original_address);
-  if (!last_lookup.IsLsdaValid()) {
-    return;
-  }
-
   if (metadata_list.empty()
-      || metadata_list.back().encoding != last_lookup.encoding
-      || metadata_list.back().lsda_address != last_lookup.lsda_address) {
-    metadata_list.push_back(Metadata(last_lookup.encoding, last_lookup.lsda_address,
+      || metadata_list.back().encoding != last_lookup.encoding) {
```

Comment:
We no longer care if encodings are different as long as personality index in both encodings is the same.